### PR TITLE
WT-8024 Add link text to cross-references in Arch Guide

### PR DESCRIPTION
**Pull request by keitharnoldsmith**
Update architecture guide pages so that references between them use link text, rather than defaulting to the title of each page.  Reworded text in some places to better integrate the references, often by providing the link in running text, rather than in a "See such-and-such" parenthetical.

### CODE REVIEW

File: src/docs/arch-log-file.dox [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*! @arch_page arch-log-file Log File Format
 
 This page assumes familiarity with the logging subsystem. Please see
-@ref arch-logging for the logging subsystem architecture description.
+the @ref arch-logging "Logging" architecture for a description of that subsystem.
```
Comment (ddanderson) [inline src/docs/arch-log-file.dox:4]: "the Logging architecture" feels awkward, maybe just Please see "Logging"

File: src/docs/arch-hs.dox [hunk 2]
```diff
@@ -252,12 +253,12 @@ visibility based on transaction and timestamp based visibility rules. In case th
 visible in the history store table, \c WT_NOTFOUND error is returned to the reader.
 
 @section arch_hs_rts History store and rollback-to-stable
-Rollback-to-stable searches the history store table for valid updates that are stable according to
```
Comment (ddanderson) [inline src/docs/arch-hs.dox:255]: It is simply "rollback to stable" (no hyphens) on that page and elsewhere.  Could you fix it in this page please?

File: src/docs/arch-logging.dox [hunk 3]
```diff
@@ -167,8 +167,7 @@ in the case of crash or failure. When the system is restarted, WiredTiger will a
 run recovery. If the system previously shutdown cleanly, then recovery determines that
 the log does not need to be replayed. Otherwise, recovery replays all operations from
 the LSN of the most recent checkpoint. When timestamps are in use, recovery also
-performs a rollback-to-stable operation on tables that are not logged.
-This aspect of recovery is discussed in @ref arch-rts.
+performs a @ref arch-rts "rollback-to-stable" operation on tables that are not logged.
```
Comment (ddanderson) [inline src/docs/arch-logging.dox:170]: "rollback to stable" (no hyphens).

**Review by ddanderson — APPROVED** [review]
LGTM made a few minor comments.
Thank you!

File: src/docs/arch-log-file.dox (hunk 1 above)
Comment (keitharnoldsmith) [reply src/docs/arch-log-file.dox]: Done

File: src/docs/arch-hs.dox (hunk 2 above)
Comment (keitharnoldsmith) [reply src/docs/arch-hs.dox]: Done.  Fixed all of the "rollback-to-stable" appearances on this page and grep'ed the rest of the architecture guide to identify others.

File: src/docs/arch-logging.dox (hunk 3 above)
Comment (keitharnoldsmith) [reply src/docs/arch-logging.dox]: Fixed

**Pull request merged**